clippy_lints/src/dereference.rs

```diff
@@ -728,7 +728,19 @@ fn deref_method_same_type<'tcx>(result_ty: Ty<'tcx>, arg_ty: Ty<'tcx>) -> bool {
     }
 }
 
-fn in_postfix_position<'tcx>(cx: &LateContext<'tcx>, e: &'tcx Expr<'tcx>) -> bool {
+fn is_in_method_chain<'tcx>(cx: &LateContext<'tcx>, e: &'tcx Expr<'tcx>) -> bool {
+    if let ExprKind::MethodCall(_, child, _, _) = e.kind {
+        match child.kind {
+            ExprKind::MethodCall(..)
+            | ExprKind::Index(..)
+            | ExprKind::Match(.., MatchSource::TryDesugar(..) | MatchSource::AwaitDesugar)
+            | ExprKind::Field(..) => {
```

Comment:
Why all those? This does not correspond to the lint description, this must be discussed if you want to increase the number of situations where the lint does not trigger.